### PR TITLE
feat(condo): DOMA-9296 make cross-domain test utils

### PR DESCRIPTION
**Pull request by dkoviazin**
For cross-domain tests
Added recursive dependancies calculation for utils

### CODE REVIEW

**Review by VKislov — APPROVED** [review]

**Comment by VKislov** [comment]
thanks to this pull request I learned what mixins are) LGTM 
(* ^ ω ^)

File: apps/condo/domains/billing/utils/testSchema/testUtils.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                    mixin.dependsOn.forEach(addMixinToGraph)
+                }
+            }
+        }
+        mixins.forEach(addMixinToGraph)
+        const sortedMixins = this.topologicalSort(dependencyGraph)
+        return sortedMixins.map(mixin => resolvedMixins.get(mixin))
+    }
+
+    topologicalSort (dependencyGraph) {
```
Comment (sitozzz) [inline apps/condo/domains/billing/utils/testSchema/testUtils.js:57]: It'll be better to mark this one as "private method"

File: apps/condo/domains/billing/utils/testSchema/testUtils.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    constructor (mixins = []) {
+        this.clients = {}
+        const resolvedMixins = this.resolveMixins(mixins)
+        if (resolvedMixins.length) {
+            Object.assign(this, ...resolvedMixins)
+        }
+        this.mixins = resolvedMixins
+    }
+
+    resolveMixins (mixins) {
```
Comment (sitozzz) [inline apps/condo/domains/billing/utils/testSchema/testUtils.js:40]: Seems to me that's private method too

File: apps/condo/domains/billing/utils/testSchema/testUtils.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            service: await makeClientWithServiceUser(),
+            admin: await makeLoggedInAdminClient(),
+            employee: {},
+        }
+        for (const mixin of this.mixins) {
+            mixin.initMixin && await mixin.initMixin.call(this)
+        }
+    }
+
+    randomNumber (numDigits) {
```
Comment (sitozzz) [inline apps/condo/domains/billing/utils/testSchema/testUtils.js:109]: Strange place for that helper. I don't think this is the right place to store random number generation function inside factory class, which intended for test utils dependency resolving (if I understand the solution architecture correctly)

File: apps/condo/domains/billing/utils/testSchema/testUtils.js (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/billing/utils/testSchema/testUtils.js]: Done

File: apps/condo/domains/billing/utils/testSchema/testUtils.js (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/billing/utils/testSchema/testUtils.js]: Donw

File: apps/condo/domains/billing/utils/testSchema/testUtils.js (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/billing/utils/testSchema/testUtils.js]: Done

**Review by sitozzz — APPROVED** [review]

**Pull request merged**